clippy_lints/src/if_let_some_result.rs

```diff
@@ -39,19 +41,37 @@ declare_lint_pass!(OkIfLet => [IF_LET_SOME_RESULT]);
 impl<'a, 'tcx> LateLintPass<'a, 'tcx> for OkIfLet {
     fn check_expr(&mut self, cx: &LateContext<'a, 'tcx>, expr: &'tcx Expr<'_>) {
         if_chain! { //begin checking variables
-            if let ExprKind::Match(ref op, ref body, ref source) = expr.kind; //test if expr is a match
-            if let MatchSource::IfLetDesugar { .. } = *source; //test if it is an If Let
-            if let ExprKind::MethodCall(_, _, ref result_types) = op.kind; //check is expr.ok() has type Result<T,E>.ok()
+            if let ExprKind::Match(ref op, ref body, source) = expr.kind; //test if expr is a match
+            if let MatchSource::IfLetDesugar { .. } = source; //test if it is an If Let
+            if let ExprKind::MethodCall(_, ok_span, ref result_types) = op.kind; //check is expr.ok() has type Result<T,E>.ok()
             if let PatKind::TupleStruct(QPath::Resolved(_, ref x), ref y, _)  = body[0].pat.kind; //get operation
             if method_chain_args(op, &["ok"]).is_some(); //test to see if using ok() methoduse std::marker::Sized;
 
             then {
                 let is_result_type = match_type(cx, cx.tables.expr_ty(&result_types[0]), &paths::RESULT);
-                let some_expr_string = snippet(cx, y[0].span, "");
+                let mut applicability = Applicability::MachineApplicable;
+                // ok_span = `ok`
+                // op.span = `x.parse()   .   ok()`
+                // op.span.until(op.span.with_lo(ok_span.lo() - BytePos(1))) = `x.parse()   .`
+                // op.span.with_lo(ok_span.lo() - BytePos(1)) = ` ok()`
+                // op.span.with_hi(ok_span.hi() - BytePos(1)) = `x.parse()   .   o`
```

Comment:
Ah, `ok_span` is just `ok`, not `ok()`. The last formatting weirdness, that will now break the suggestion would be 

```rust
if let Some(_) = x.ok ()
```

We know at this point, that the method call is `_.ok()` and not trailed by some other method call like `_.ok().and_then()`, right? In that case we could use the `op.span.hi()` instead of adding `BytePos(2)` to the `ok_span.hi()`.

---

Sorry for my annoying weird formatting nitpicking

Reply:
Addressed. Don't mind, it's important to prevent wrong suggestions :)